### PR TITLE
SYS-1210: Easier editing of file-level item sequence numbers

### PR DESCRIPTION
**Pull request by ztucker4**
Implements [SYS-1210](https://jira.library.ucla.edu/browse/SYS-1210)

On interview-level items only, adds a link ("Configure file order") to a new template (`order_files`) containing a form allowing users to edit sequence numbers on child ProjectItems.

This form does not enforce any new validation on sequence numbers, i.e. users can still enter duplicate sequence numbers, numbers < 1, etc. Changing a sequence number via this form will also update the modified date and modified by fields on the associated item. 

Screenshots:
image
---
image

### CODE REVIEW

**Review by akohler — CHANGES_REQUESTED** [review]
@ztucker4 This all works very well - the functionality is perfect.  You did a great job of figuring out how to do this, and as far as I can tell, this is the best - maybe the only - way to do this.

There are a couple of minor code changes, but most of my comments are about documentation.  In order to understand this, I had to do a lot of the same research you probably did.  Please help future-us-and-other-maintainers by adding comments which fill in some of those blanks.   (And feel free to ding me on the same thing when reviewing my PRs!  I find myself improving comments I wrote which confuse me when revisited...)

Thanks --Andy

File: oh_staff_ui/views_utils.py [hunk 1]
```diff
@@ -367,3 +369,34 @@ def get_relatives(item: ProjectItem) -> dict:
...
+
+
+def get_sequence_formset(items: list) -> BaseFormSet:
+    sequence_list = []
+    for item in items:
+        sequence_list.append({"item": item, "sequence": item.sequence})
+    factory = formset_factory(
+        ItemSequenceForm, extra=0, formset=BaseItemSequenceFormset, validate_min=True
+    )
+    formset = factory(initial=sequence_list, form_kwargs={"items_list": items})
```
[the remark this answers is not in the document]
Comment (ztucker4) [reply oh_staff_ui/views_utils.py:383]: Added comments here and in `save_sequence_data`. I realized that the `item`s in `sequence_list` were never actually used, so I've removed them.

File: oh_staff_ui/views_utils.py [hunk 2]
```diff
@@ -367,3 +369,34 @@ def get_relatives(item: ProjectItem) -> dict:
...
+        for form in formset:
+            if form.is_valid():
+                sequence = form.cleaned_data["sequence"]
+                child_item = form.item
+                child_item.sequence = sequence
+                child_item.last_modified_date = timezone.now()
+                child_item.last_modified_by = request.user
+                child_item.save()
+        messages.success(request, "Sequence numbers have been saved.")
+    else:
```
[the remark this answers is not in the document]
Comment (ztucker4) [reply oh_staff_ui/views_utils.py:403]: No, I don't know any way to trigger an invalid form here currently.

**Review by akohler — APPROVED** [review]

**Comment by akohler** [comment]
@ztucker4 Thanks for the comment improvements, these help add context and are clear enough now.  Merging.

**Pull request merged**